src/components/DendogramVis.js

```diff
@@ -0,0 +1,138 @@
+import React, { Component } from 'react';
+import * as d3 from 'd3';
+import { useD3 } from '../hooks/useD3';
+import '../App.css';
+import { keys } from '@material-ui/core/styles/createBreakpoints';
+
+// var data = {"children":
+// [{"name":"boss1","children":[{"name":"mister_a","colname":"level3", "color": "red"},{"name":"mister_b","colname":"level3"},{"name":"mister_c","colname":"level3"},{"name":"mister_d","colname":"level3"}],"colname":"level2"},{"name":"boss2","children":[{"name":"mister_e","colname":"level3"},{"name":"mister_f","colname":"level3"},{"name":"mister_g","colname":"level3"},{"name":"mister_h","colname":"level3"}],"colname":"level2"}],"name":"CEO"};
+  
+
+// Purpose: ANnotate with some meta-data useful for visualization
+function enrichDendogram(node, color) {
```

Comment:
To enrich the dendogram output to be suitable for visualization purposes.

For example, input data only has the root of each sub-tree marked with color, then we will propogate this color function down to other nodes in same sub-tree